src/compiler/checker.ts

```diff
@@ -1793,6 +1793,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (type === emptyGenericType) return undefined;
             return type;
         },
+        createTypeReference: (target, typeArguments) => {
+            if (target !== emptyGenericType && getObjectFlags(target) & ObjectFlags.ClassOrInterface) {
+                const interfaceType = target as InterfaceType;
+                if (some(interfaceType.typeParameters)) {
+                    return createTypeReference(interfaceType as GenericType, typeArguments);

```

Comment:
It's up to the caller of `createTypeReference` to validate that the `typeArguments` have an appropriate count, and to fill missing values with defaults - should we not have a `fillMissingTypeArguments` and some count checks in here?

Reply:
Alternatively, maybe just exposing a `getAnyAsyncIterator` might be more purpose-built for the quickfix, and we wouldn't have to harden it against misuse, since that's the one thing it's used for.